reference/api/tasks.md

```diff
@@ -262,3 +272,59 @@ Get a single task.
 ## Cancel tasks
 
 ## Delete tasks
+
+<RouteHighlighter method="DELETE" route="/tasks?{task_uid}"/>
+
+Delete a finished (`succeeded`, `failed`, or `canceled`) task based on `uid`, `status`, `type`, `indexUid`, `canceledBy`, or date. Task deletion is an atomic transaction, **either all tasks are successfully deleted, or none are**.
+
+::: warning
+Using this route without any filters (DELETE `/tasks`) will result in the `missing_task_filters` error. This error prevents users from accidentally deleting the entire history.
+:::
+
+### Query parameters
+
+A valid `uids`, `statuses`, `types`, `indexUids`, `canceledBy`, or date(`beforeXAt` or `afterXAt`) field is required.
+
+| Query Parameter        | Description                                                                                                                          |
+| :--------------------- | :----------------------------------------------------------------------------------------------------------------------------------- |
+| **`uids`**             | Delete tasks based on `uid`. Separate multiple `uids` with a comma (`,`). Use `uids=*` for all `uids`                                |
+| **`statuses`**         | Delete tasks based on `status`. Separate multiple `statuses` with a comma (`,`). Use `statuses=*` for all `statuses`                 |
+| **`types`**            | Delete tasks based on `type`. Separate multiple `types` with a comma (`,`). Use `types=*` for all `types`                            |
+| **`indexUids`**        | Delete tasks based on `indexUid`. Separate multiple `uids` with a comma (`,`). Use `indexUids=*` for all `indexUids`. Case-sensitive |
+| **`canceledBy`**       | Delete tasks based on the `canceledBy` field                                                                                         |
+| **`beforeEnqueuedAt`** | Delete tasks before a specified `enqueuedAt` date                                                                                    |
+| **`beforeStartedAt`**  | Delete tasks before a specified `startedAt` date                                                                                     |
+| **`beforeFinishedAt`** | Delete tasks before a specified `finishedAt` date                                                                                    |
+| **`afterEnqueuedAt`**  | Delete tasks after a specified `enqueuedAt` date                                                                                     |
+| **`afterStartedAt`**   | Delete tasks after a specified `startedAt` date                                                                                      |
+| **`afterFinishedAt`**  | Delete tasks after a specified `finishedAt` date                                                                                     |
```

Comment:
Do you think we should specify the fact that these filters are all exclusive?
For example, if I ask for every task `beforeStartedAt=2022-11-21` then I'll get all the tasks that happened before this day.
And if I ask all the tasks `afterStartedAt=2022-11-21`, then I'll get all the tasks AFTER this day, from the first nanosecond of 2022-11-22.

To me, this seems logical, so maybe it's not worth mentioning it, but I know when implementing the features that some people felt like the before should be exclusive and the after should be inclusive (e.g.: `afterStartedAt=2022-11-21` would include the tasks that happened during my birthday (the 21)).

Reply:
I mentioned this in the [guide](https://github.com/meilisearch/documentation/blob/808bdc4cc348e43efcf5bf6d8e0eb569af3f78cc/learn/advanced/asynchronous_operations.md#filter-by-date), but I can add a note here too!